### PR TITLE
lib(easy-peasy): Upgrade easy-peasy to v6; replace custom persist implementation with new upstream version

### PR DESCRIPTION
**Pull request by damassi**
This PR resolves [MOPLAT-744]

### Description

This upgrades our state lib from v3 to v6, and addresses some breaking changes: 

- No more `initialData` on the `createContextStore` [provider](https://easy-peasy.vercel.app/docs/api/create-context-store.html), in favor of `runtimeModel`.  
- No more `createComponentStore`, in favor of the `useLocalStore` [hook](https://github.com/ctrlplusb/easy-peasy/releases/tag/v4.0.0). 

Also swaps out our persist implementation with the new version that comes with easy-peasy 6, and deletes old code (including legacy migration tests that no longer apply). 

Still need to double check that things are all working in the app, but otherwise this is ready to review! 

cc @artsy/mobile-platform 

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-744]: https://artsyproduct.atlassian.net/browse/MOPLAT-744?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/store/GlobalStore.tsx [hunk 1]
```diff
@@ -2,59 +2,76 @@ import { __unsafe_mainModalStackRef } from "app/NativeModules/ARScreenPresenterM
...
-      __globalStoreTestUtils__.dispatchedActions.push(_action)
-      next(_action)
+
+    middleware.push((_api) => (next) => async (action) => {
+      if (action.type) {
+        __globalStoreTestUtils__.dispatchedActions.push(action)
+      }
+
+      const result = next(action)
+      return result
```
Comment (damassi) [inline src/app/store/GlobalStore.tsx:47]: This was the issue with the missing data returned from the thunk; our custom middleware wasn't returning anything

File: src/app/store/__tests__/persistence.tests.ts [hunk 2]
```diff
@@ -122,133 +113,3 @@ describe("assignDeep", () => {
     expect(obj).toEqual({ foo: [{}] })
   })
 })
-
-describe(persist, () => {
```
Comment (damassi) [inline src/app/store/__tests__/persistence.tests.ts:126]: don't need any of these tests anymore

File: src/app/store/GlobalStore.tsx (hunk 1 above)
Comment (gkartalis) [reply src/app/store/GlobalStore.tsx:47]: 🤯

File: src/app/store/__tests__/persistence.tests.ts (hunk 2 above)
Comment (gkartalis) [reply src/app/store/__tests__/persistence.tests.ts:126]: so nice to see code removed 🔥

File: src/app/store/migration.ts [hunk 3]
```diff
@@ -1,300 +1,18 @@
...
-  AddDirtyFormValuesToSubmissionState: 28,
-  RemoveDeviceId: 29,
-  AddSubmissionIdForMyCollection: 30,
-  AddRecentPriceRangesModel: 31,
-  AddUserPreferredPriceRange: 32,
-  RenameAdminToLocalOverridesForFeatures: 33,
-  MoveEnvironmentToDevicePrefsAndRenameAdminToLocal: 34,
-  AddCategoryToSubmissionArtworkDetails: 35,
-  AddOnboardingArtQuizStateToAuthModel: 36,
+  MigrateToUpstreamPersistedState: 1,
```
Comment (gkartalis) [inline src/app/store/migration.ts:7]: 👏

**Pull request closed without merging**